test/access/Ownable.test.js

```diff
@@ -1,72 +1,79 @@
-const { constants, expectEvent } = require('@openzeppelin/test-helpers');
-const { expectRevertCustomError } = require('../helpers/customError');
-
-const { ZERO_ADDRESS } = constants;
-
+const { ethers } = require('hardhat');
 const { expect } = require('chai');
+const { loadFixture } = require('@nomicfoundation/hardhat-network-helpers');
 
-const Ownable = artifacts.require('$Ownable');
-
-contract('Ownable', function (accounts) {
-  const [owner, other] = accounts;
+async function fixture() {
+  const accounts = await ethers.getSigners(); // this is slow :/
+  const owner = accounts.shift();
+  const other = accounts.shift();
+  const ownable = await ethers.deployContract('$Ownable', [owner]);
+  return { accounts, owner, other, ownable };
+}
 
+describe('Ownable', function () {
   beforeEach(async function () {
-    this.ownable = await Ownable.new(owner);
+    await loadFixture(fixture).then(results => Object.assign(this, results));
```

Comment:
[This documentation](https://hardhat.org/hardhat-runner/docs/guides/test-contracts) uses await for doing `loadFixture`. Considering the fixture itself is async, it makes a lot of sens that loading it would be async.

As for the syntax, I have a slight preference for my version, but I'd be ok with your version as well. (If I had to be specific, what ont thing I like about mine is that the first thing is the await, so it is clear that the line is async. When the await is in the middle, its harder to identify that this line requires the function to be async).

Reply:
To be clear, I don't have a strong preference between `.then` and `await`, but I think the construction of this function is problematic because `loadFixture` returns a promise, and you can solve a promise either with `.then` or with `await` but not both.

In this case, `loadFixture(...).then(...)` will return an object, so the first `await` shouldn't be needed. However, the function is `async` so we need to make sure the `beforeEach` will solve the promise before starting the tests. My conclusion is that this line works, but it works in a very strange way. Let's follow the [Mocha guidelines for promises](https://mochajs.org/#working-with-promises) and pick just one between `await` and `.then` and be consistent:

```js
  beforeEach(function () {
    return loadFixture(fixture).then(results => Object.assign(this, results));
  })
```

We can keep the `await` version in 2 parts to avoid having the `await` in the middle:

```js
beforeEach(async function () {
  const results = await loadFixture(fixture);
  Object.assign(this, results);
})
```

Reply:
> Using then doesn't resolve a promise, it transform as promise (here that the fixture is loaded)

It solves it in the `.then()` callback. The point is that is confusing to use `.then` but then `await`ing for the returned promise to be solved. The `beforeEach` will already do that if we return such function:

```js
beforeEach(function () {
    return loadFixture(fixture).then(results => Object.assign(this, results));
})
```

Generally, if we'll use `await` anyways, I'd rather avoid `.then`